processor/k8sprocessor/factory.go

```diff
@@ -191,3 +194,26 @@ func createProcessorOpts(cfg config.Processor) []Option {
 
 	return opts
 }
+
+func warnDeprecatedMetadataConfig(logger *zap.Logger, cfg config.Processor) {
+	oCfg := cfg.(*Config)
+
+	for _, field := range oCfg.Extract.Metadata {
+		switch field {
+		case metdataNamespace:
+			logger.Warn("metdataNamespace is a deprecated configuration for k8s-tagger processor")
+		case metadataPodName:
+			logger.Warn("metadataPodName is a deprecated configuration for k8s-tagger processor")
+		case metadataPodUID:
+			logger.Warn("metadataPodUID is a deprecated configuration for k8s-tagger processor")
+		case metadataStartTime:
+			logger.Warn("metadataStartTime is a deprecated configuration for k8s-tagger processor")
+		case metadataDeployment:
+			logger.Warn("metadataDeployment is a deprecated configuration for k8s-tagger processor")
+		case metadataCluster:
+			logger.Warn("metadataCluster is a deprecated configuration for k8s-tagger processor")
+		case metadataNode:
+			logger.Warn("metadataNode is a deprecated configuration for k8s-tagger processor")
+		}
```

Comment:
This block could be re-written as:

```go
var oldName, newName string

switch field {
      case metadataNamespace:
            oldName = metadataNamesapce
            newName = conventions.AttributesK8SNamespace

      // other cases here
}

if oldName != "" {
    logger.Warn("warning here")
}
```

or you could use a Map instead of switch case. Since this will run only once on startup, I think map is fine if it simplifies the function. Up to you.